src/drive/web/modules/views/Folder/FolderViewBody.jsx

```diff
@@ -49,27 +51,42 @@ const FolderViewBody = ({
     <>
       <SelectionBar actions={actions} />
       <FileListv2>
-        <MobileFileListHeader
-          folderId={null}
-          canSort={canSort}
-          sort={sortOrder}
-          onFolderSort={changeSortOrder}
-          thumbnailSizeBig={isBigThumbnail}
-          toggleThumbnailSize={toggleThumbnailSize}
-        />
-        <FileListHeader
-          folderId={null}
-          canSort={canSort}
-          sort={sortOrder}
-          onFolderSort={changeSortOrder}
-          thumbnailSizeBig={isBigThumbnail}
-          toggleThumbnailSize={toggleThumbnailSize}
-        />
+        {hasDataToShow && (
+          <>
+            <MobileFileListHeader
+              folderId={null}
+              canSort={canSort}
+              sort={sortOrder}
+              onFolderSort={changeSortOrder}
+              thumbnailSizeBig={isBigThumbnail}
+              toggleThumbnailSize={toggleThumbnailSize}
+            />
+            <FileListHeader
+              folderId={null}
+              canSort={canSort}
+              sort={sortOrder}
+              onFolderSort={changeSortOrder}
+              thumbnailSizeBig={isBigThumbnail}
+              toggleThumbnailSize={toggleThumbnailSize}
+            />
+          </>
+        )}
         <FileListBodyV2 selectionModeActive={false}>
           <AddFolder />
           {isInError && <Oops />}
           {isLoading && <FileListRowsPlaceholder />}
-          {isEmpty && <EmptyDrive canUpload={true} />}
+          {/* TODO FolderViewBody should not have the responsability to chose 
+          which empty component to display. It should be done by the "view" itself. 
+          But adding a new prop like <FolderViewBody emptyComponent={} 
+          is not good enought too */}
```

Comment:
Why is the emptyComponent prop not good ? 

s/enought/enough
s/too/? (not grammatically correct)